File: Finance/percentage.py
class product:
    def __init__(self, OriginalPrice: float, AfterPrice: float) -> None:
        self.OriginalPrice = OriginalPrice
        self.AfterPrice = AfterPrice
        self.PriceDiff = self.AfterPrice - self.OriginalPrice

    def percentage(self) -> float:
        return round(
            abs((NeoProduct.OriginalPrice - NeoProduct.AfterPrice) / self.OriginalPrice)
            * 100,
            2,
        )


NeoProduct = product(float(input("Preço original: ")), float(input("Preço agora: ")))
if NeoProduct.PriceDiff < 0:
    print(f"Desconto de {NeoProduct.percentage()}%.")
elif NeoProduct.PriceDiff > 0:
    print(f"Juros de {NeoProduct.percentage()}%.")
else:
    print("Mesmo preço!")

File: Finance/percentage.cpp
#include <iostream>

int main()
{
    double initialPrice, finalPrice;
    std::cout << "Preço inicial: ";
    std::cin >> initialPrice;
    std::cout << "Preço final: ";
    std::cin >> finalPrice;
    double PriceDiff = finalPrice - initialPrice;

    double percentage = (abs(initialPrice - finalPrice) / initialPrice) * 100;

    if (PriceDiff > 0) // juros
    {
        std::cout << "Juros de ";
    }
    else if (PriceDiff < 0) // desconto
    {
        std::cout << "Desconto de ";
    }
    else // não houve alteração
    {
        std::cout << "Não houve alteração de preço!";
    }
    std::cout << percentage << "%." << std::endl;

    return 0;
}
